### PR TITLE
[INF-2535] Update to use AWS SDK v3

### PR DESCRIPTION
**Pull request by ohookins**
Much like the previous PRs in the other codebases, although this codebase has already largely been modernised to support async/await around the place so there isn't nearly as much bridging to be done between the AWS SDK v3 with Promises and callbacks in the rest of the codebase, fortunately.

Again, most of the changes are just in the tests to support the newer style of mocking the SDK. One thing I haven't been able to figure out is the before/after mock resets. There is a test ordering issue that seems to persist unless I reset the mocks both before and after each test, and while it seems like `afterEach` should only be required (and is indeed being called properly), some test ordering issues persisted. I'm leaving resets in both before and after functions for the moment unless someone can point out what is wrong.

### CODE REVIEW

**Review by andrewmcnamara — COMMENTED** [review]
I'm looking a. this `EventEmitter` pattern and wondering if  a Generator function would  simplify

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*

At the end of the day isn't just  a state machine ?

https://gist.github.com/getify/83e1f7072cae4d60bfbe

**Comment by ohookins** [comment]
Well yes, but the state machine is actually in AWS and we are just collecting what is happening there, and sending a "simplified" version of it to the parts of the codebase that are interested in it.

I'm trying to avoid making big refactors though, and just massage the code into the shape necessary to update the AWS SDK.

**Comment by ohookins** [comment]
Have tested locally and it works as expected both for successful and failing deploys.

File: lib/evaluators/draining.js [hunk 1]
```diff
@@ -1,6 +1,4 @@
-'use strict'
-
-const AWS = require('aws-sdk');
+'use strict';
 const _ = require('lodash');
```
[the remark this answers is not in the document]
Comment (ohookins) [reply lib/evaluators/draining.js:2]: Yes, if I actually do that. It's a lot more pervasive in this codebase and not all of the usage has a nice ES6 equivalent.

**Review by sushmithaeshwar — APPROVED** [review]
LGTM

**Pull request merged**